### PR TITLE
Create Bucket with IA/Archive support and add restore API

### PR DESCRIPTION
**Pull request by qixu001**
1 Supports IA/Archive in CreateBucket API.
  ---The original CreateBucket does not have the request body, which does not fully implement the PUT Bucket API spec.

2 Add Restore API.

### CODE REVIEW

File: sdk/Enums/StorageClass.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ */
+using System;
+namespace Aliyun.OSS
+{
+    public enum StorageClass
+    {
+        Standard,
+        IA,
+        Archive
+    }
```
Comment (baiyubin) [inline sdk/Enums/StorageClass.cs]: - 请添加注释简单说明一下三种存储类型。

File: sdk/IOss.cs [hunk 2]
```diff
@@ -656,6 +665,14 @@ public interface IOss
         /// <returns><see cref="AccessControlList" /> instance</returns>
         AccessControlList GetObjectAcl(string bucketName, string key);
 
+        /// <summary>
+        /// Restores the object.
+        /// </summary>
+        /// <returns>The object.</returns>
+        /// <param name="bucketName">Bucket name.</param>
+        /// <param name="key">Key.</param>
+        RestoreObjectResult RestoreObject(string bucketName, string key);
```
Comment (baiyubin) [inline sdk/IOss.cs]: - RestoreObject需要在 aliyun-oss-csharp-sdk\samples\Samples 添加使用示例。

File: sdk/OssClient.cs [hunk 3]
```diff
@@ -193,11 +194,17 @@ public void SetEndpoint(Uri endpoint)
 
         /// <inheritdoc/>
         public Bucket CreateBucket(string bucketName)
+        {
+            return CreateBucket(bucketName, StorageClass.Standard);
```
Comment (baiyubin) [inline sdk/OssClient.cs]: - 调用 CreateBucket(bucketName, null);比较合适。

File: sdk/OssClient.cs [hunk 4]
```diff
@@ -1054,6 +1061,19 @@ public AccessControlList GetObjectAcl(string bucketName, string key)
...
+        /// <inheritdoc/>
+        public RestoreObjectResult RestoreObject(string bucketName, string key)
+        {
+            ExecutionContext context = CreateContext(HttpMethod.Post, bucketName, key);
+            var cmd = RestoreObjectCommand.Create(_serviceClient,
+                                                 _endpoint,
+                                                  context,
+                                                 bucketName,
+                                                  key);
+
```
Comment (baiyubin) [inline sdk/OssClient.cs]: - RestoreObjectCommand.Create 的几个参数对齐。

File: sdk/Commands/CreateBucketCommand.cs [hunk 5]
```diff
@@ -24,19 +26,35 @@ protected override string Bucket
...
         {
             OssUtils.CheckBucketName(bucketName);
             _bucketName = bucketName;
+            StorageClass = storageClass;
         }
 
         public static CreateBucketCommand Create(IServiceClient client, Uri endpoint,
                                                  ExecutionContext context,
-                                                 string bucketName)
+                                                 string bucketName, StorageClass storageClass = StorageClass.Standard)
```
[the remark this answers is not in the document]
Comment (qixu001) [reply sdk/Commands/CreateBucketCommand.cs]: ok

File: sdk/Enums/StorageClass.cs (hunk 1 above)
Comment (qixu001) [reply sdk/Enums/StorageClass.cs]: ok

File: sdk/OssClient.cs (hunk 3 above)
Comment (qixu001) [reply sdk/OssClient.cs]: ok

File: sdk/Transform/CreateBucketRequestSerializer.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        }
+
+        public override Stream Serialize(StorageClass request)
+        {
+            var model = new CreateBucketRequestModel
+            {
+                StorageClass = request
+            };
+
+            return ContentSerializer.Serialize(model);
```
[the remark this answers is not in the document]
Comment (qixu001) [reply sdk/Transform/CreateBucketRequestSerializer.cs]: 我建议在调用的地方判断如果StorageClass？是null直接返回null。CreateBucketRequestSerializer.cs里还是返回正常的Stream比较好。

File: sdk/OssClient.cs (hunk 4 above)
Comment (qixu001) [reply sdk/OssClient.cs]: done

File: sdk/Transform/CreateBucketRequestSerializer.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (baiyubin) [reply sdk/Transform/CreateBucketRequestSerializer.cs]: ok

**Pull request closed without merging**